networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2Network.java

```diff
@@ -245,7 +245,7 @@ public synchronized void stop() {
     proposerSlashingGossipManager.shutdown();
     attesterSlashingGossipManager.shutdown();
     attestationSubnetService.unsubscribe(discoveryNetworkAttestationSubnetsSubscription);
-    super.stop();
+    return peerManager.sendGoodbyeToPeers().thenCompose(__ -> super.stop());
```

Comment:
Probably always want to call `super.stop` now I think about this, even if sending goodbye fails.  So probably:
```suggestion
    return peerManager.sendGoodbyeToPeers().exceptionally(error -> {
      LOG.debug("Failed to send goodby to peers on shutdown", error);
      return null;
    }).thenCompose(__ -> super.stop());
```

Reply:
Done.